### PR TITLE
Refactor and rename Project module to Recipes for clarity, fix bug in `load_projects`

### PR DESCRIPTION
**Pull request by shelbyspees**
I made a separate branch and rebased so that the dependency tree stuff will be in a separate PR from this, because I was getting kind of lost.

- [x] Create a `Projects` module containing `Project` and `Package` classes
- [x] Convert `Project(s)` to `Recipe(s)` and `Package` to `Target`
- [x] Make sure `Recipes.load_recipes` rspec test passes for rpm-packages (tested in local env)

### CODE REVIEW

File: examples/golang/recipe.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class Golang < FPM::Cookery::Recipe
 
-  name        'grindr-golang'
+  name        'golang'
```
Comment (smaslennikov) [inline examples/golang/recipe.rb]: This should probably stay with the `grindr-` prefix, unless you have a reason not to

Comment (shelbyspees) [reply examples/golang/recipe.rb]: The dependency matcher goes based on the name assigned here in the recipe. I could rename everything to `grindr-golang` but it was simpler to just change it in this one spot.

File: spec/recipes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        path = Dir.pwd
+        expect(Recipes.inspect(path)).to be nil
+      end
+    end
+  end
+
+  describe '.load_recipes' do
+    context 'when passed a valid directory containing recipes' do
+      before(:all) do
+        @recipes = Recipes.load_recipes('recipes/recipes/')
```
Comment (shelbyspees) [inline spec/recipes_spec.rb]: @smaslennikov Can we update vagrant so that the file path here makes a little more sense?

Comment (smaslennikov) [reply spec/recipes_spec.rb]: @naftulikay ^ this is your change

**Pull request merged**